src/services/featurehelper.js

```diff
@@ -446,6 +450,104 @@ ngeo.FeatureHelper.prototype.getStrokeProperty = function(feature) {
 };
 
 
+// === EXPORT ===
+
+
+/**
+ * Export features in the given format. The projection of the exported features
+ * is: `EPSG:4326`.
+ * @param {Array.<ol.Feature>} features Array of vector features.
+ * @param {string} formatType Format type to export the features.
+ * @export
+ */
+ngeo.FeatureHelper.prototype.export = function(features, formatType) {
+  switch (formatType) {
+    case ngeo.FeatureHelper.FormatType.GPX:
+      this.exportGPX(features)
+      break;
+    case ngeo.FeatureHelper.FormatType.KML:
+      this.exportKML(features)
+      break;
+    default:
+      break;
+  }
+};
+
+
+/**
+ * Export features in GPX and download the result to the browser. The
+ * projection of the exported features is: `EPSG:4326`.
+ * @param {Array.<ol.Feature>} features Array of vector features.
+ * @export
+ */
+ngeo.FeatureHelper.prototype.exportGPX = function(features) {
+  var format = new ol.format.GPX();
+  var mimeType = 'application/gpx+xml';
+  var fileName = 'export.gpx';
+  this.export_(features, format, fileName, mimeType);
+};
+
+
+/**
+ * Export features in KML and download the result to the browser. The
+ * projection of the exported features is: `EPSG:4326`.
+ * @param {Array.<ol.Feature>} features Array of vector features.
+ * @export
+ */
+ngeo.FeatureHelper.prototype.exportKML = function(features) {
+  var format = new ol.format.KML();
+  var mimeType = 'application/vnd.google-earth.kml+xml';
+  var fileName = 'export.kml';
+  this.export_(features, format, fileName, mimeType);
+};
+
+
+/**
+ * Export features using a given format to a specific filename and download
+ * the result to the browser. The projection of the exported features is:
+ * `EPSG:4326`.
+ * @param {Array.<ol.Feature>} features Array of vector features.
+ * @param {ol.format.Feature} format Format
+ * @param {string} fileName Name of the file.
+ * @param {string=} opt_mimeType Mime type. Defaults to 'text/plain'.
+ * @private
+ */
+ngeo.FeatureHelper.prototype.export_ = function(features, format, fileName,
+    opt_mimeType) {
+  var mimeType = opt_mimeType !== undefined ? opt_mimeType : 'text/plain';
+
+  // clone the features to:
+  //  - transform the exported features to EPSG:4326
+  //  - apply the original style to the clone (the original may have select
+  //    style active)
+  var clones = [];
+  var clone;
+  features.forEach(function(feature) {
+    clone = new ol.Feature();
+    clone.setProperties(feature.getProperties());
+    clone.setGeometry(clone.getGeometry().clone());
+    if (this.projection_) {
+      clone.getGeometry().transform(this.projection_, ol.proj.get('EPSG:4326'));
+    }
+    this.setStyle(clone, false);
+    clones.push(clone);
+  }, this);
+
+  var data = format.writeFeatures(clones);
```

Comment:
For transforming to 'EPSG:4326', the above is an excellent idea, but we still need to clone the features.  The exportation may occur while a feature is selected, therefore it may have the "select" style applied (halo, vertex).

I think it's okay to leave the cloning there.